pandas/core/util/numba_.py

```diff
@@ -123,6 +109,8 @@ def prepare_function_arguments(
         user input positional arguments
     kwargs : dict
         user input keyword arguments
+    num_required_args : int
+        the number of required leading positional arguments for udf.
```

Comment:
I think this is the number of arguments that pandas (and not the user) will pass, right? So perhaps a better description is

> The number of positional arguments we will pass to udf. These are not supplied by the user.

Reply:
A detailed description is added.